terminus-ui/paginator/src/paginator.component.spec.ts

```diff
@@ -1,470 +1,309 @@
-      // Fake the change event that Angular would normally trigger
-      component.ngOnChanges(changesMock1);
-
-      expect(component.currentPageIndex).toEqual(1);
-
-      const changesMock2: SimpleChanges = {isZeroBased: new SimpleChange(false, true, false)};
-      // Fake the change event that Angular would normally trigger
-      component.ngOnChanges(changesMock2);
-
-      expect(component.currentPageIndex).toEqual(0);
-    });
-
-  });
-
-
-  describe(`currentPageChanged()`, () => {
 
-    test(`should set the current page, and trigger methods`, () => {
-      const eventMock = {
-        name: '21 - 30 of 125',
-        value: 2,
-      };
-      component.pageSelect.emit = jest.fn();
-      component['createCurrentPageLabel'] = jest.fn();
-      component.totalRecords = 125;
-      component.ngAfterViewInit();
-
-      expect(component.currentPageIndex).toEqual(0);
-
-      component.currentPageChanged(eventMock);
-
-      expect(component.currentPageIndex).toEqual(2);
-      expect(component['createCurrentPageLabel']).toHaveBeenCalled();
-      expect(component.pageSelect.emit).toHaveBeenCalled();
-    });
-
-  });
-
-
-  describe(`changePage()`, () => {
-
-    test(`should fake a changed page event when valid`, () => {
-      jest.spyOn(component, 'currentPageChanged');
-      component.totalRecords = 125;
-      component.ngAfterViewInit();
-      component.changePage(2, 1, component.pagesArray);
-
-      expect(component.currentPageChanged).toHaveBeenCalled();
-    });
-
-
-    test(`should do nothing when the destination is not valid`, () => {
-      component.totalRecords = 125;
-      component.ngAfterViewInit();
-      jest.spyOn(component, 'currentPageChanged');
-
-      component.changePage(-1, 1, component.pagesArray);
-      expect(component.currentPageChanged).not.toHaveBeenCalled();
-
-      const invalidLength = component.pagesArray.length + 2;
-      component.changePage(invalidLength, 1, component.pagesArray);
-      expect(component.currentPageChanged).not.toHaveBeenCalled();
-    });
-
-
-    test(`should do nothing when already on the requested page`, () => {
-      component.totalRecords = 125;
-      component.ngAfterViewInit();
-      jest.spyOn(component, 'currentPageChanged');
+    test(`should set and retrieve`, () => {
+      const fixture = createComponent(TestComponents.RecordsPerPage);
+      const hostComponent = fixture.componentInstance;
+      fixture.detectChanges();
 
-      component.changePage(2, 2, component.pagesArray);
-      expect(component.currentPageChanged).not.toHaveBeenCalled();
+      expect(hostComponent.showRecordsPerPageSelector).toEqual(true);
+      expect(fixture.debugElement.query(By.css('.qa-paginator-per-page-select'))).toBeTruthy();
     });
 
-  });
-
-
-  describe(`isFirstPage()`, () => {
+    test(`should change page on select`, () => {
+      const fixture = createComponent(TestComponents.RecordsPerPage);
+      const hostComponent = fixture.componentInstance;
```

Comment:
I wonder whether inside this describe scope, `fixture` and `hostComponent` could be extracted into `beforeEach` block as the exact same assignment happened so many times.

Reply:
great suggestion!